hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -677,4 +714,27 @@ private void rollbackPendingCommits() {
     });
     return compactionInstantTimeOpt;
   }
+
+  /**
+   * Auto Clean service running concurrently.
+   */
+  private static class AutoCleanerService extends AbstractAsyncService {
```

Comment:
lets move this to its own class?

Reply:
And even the `spawnAutoCleanerIfEnabled` and `waitForAutoCleanerToShutdown` as static helpers in this class.. (trying to avoid code creep inside HoodieWriteClient :))